File: software/examples/javascript/ExampleCallback.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your Motion Detector Bricklet 2.0

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var md = new Tinkerforge.BrickletMotionDetectorV2(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

// Register motion detected callback
md.on(Tinkerforge.BrickletMotionDetectorV2.CALLBACK_MOTION_DETECTED,
    // Callback function for motion detected callback
    function () {
        console.log('Motion Detected');
    }
);

// Register detection cycle ended callback
md.on(Tinkerforge.BrickletMotionDetectorV2.CALLBACK_DETECTION_CYCLE_ENDED,
    // Callback function for detection cycle ended callback
    function () {
        console.log('Detection Cycle Ended (next detection possible in ~2 seconds)');
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/ruby/example_callback.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_motion_detector_v2'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your Motion Detector Bricklet 2.0

ipcon = IPConnection.new # Create IP connection
md = BrickletMotionDetectorV2.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Register motion detected callback
md.register_callback(BrickletMotionDetectorV2::CALLBACK_MOTION_DETECTED) do
  puts 'Motion Detected'
end

# Register detection cycle ended callback
md.register_callback(BrickletMotionDetectorV2::CALLBACK_DETECTION_CYCLE_ENDED) do
  puts 'Detection Cycle Ended (next detection possible in ~2 seconds)'
end

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect
